ui/src/components/Views/Configs/MountPointForm.tsx

```diff
@@ -1,137 +1,187 @@
+      setMountPointData('../tmp');
+    }
 
-        const images = await ddClient.docker.listImages() as [DockerImage];
-        const isPresent = images.some(image => image.RepoTags?.at(0) === LATEST_IMAGE);
+    setUserState({ loading: false, selectedUser: foundUsers[0], users: foundUsers });
+  };
 
-        setHasLocalImage({ checking: false, isPresent });
+  const locateMountPoint = async () => {
+    if (ddClient.host.platform === 'win32') {
+      checkWindowsDistro();
+    } else {
+      checkUser();
+    }
+  };
 
-        if (isPresent) {
-          checkHomeDir();
-        } else {
-          setIsPullingImage(true);
-        }
+  useEffect(() => {
+    const execChecks = async () => {
+      if (userState.users.length === 0
+        || (ddClient.host.platform === 'win32' && osState.OSs.length === 0)) {
+        locateMountPoint();
       }
     };
 
     execChecks();
-  }, [triggerImageCheck]);
+  }, []);
+
+  useEffect(() => {
+    if (osState.selectedOS) {
+      checkUser();
+    }
+  }, [triggerUserCheck]);
 
   const onClose = () => {
-    setMountPointUser(userState.selectedUser);
+    setMountPointData(`${userState.selectedUser},${osState.selectedOS}`);
   };
 
-  const endOfDownloadCallback = () => {
-    setIsPullingImage(false);
-    setTriggerImageCheck(true);
+  const handleOsChange = (target: string) => {
+    setOsState({ ...osState, selectedOS: target });
+    checkUser();
   };
 
   return (
     <Dialog open onClose={onClose}>
       <DialogContent>
-        <Box >
-          <Box marginBottom={5} display="flex" gap={5} alignItems="center">
-            {hasLocalImage.checking &&
-              <Typography>
-                Checking for local LocalStack image
+        <Typography variant='h3'>
+          Default mount point settings
+        </Typography>
+        <br/>
+        <Paper sx={{ padding: 1}}>
+          {
+            ddClient.host.platform === 'win32' &&
+            <>
+              <Typography  variant='subtitle1'>
+                WSL distro
               </Typography>
-            }
-            {userState.loading &&
-              <Typography>
-                Checking for users
+              <Typography variant='body2' >
+                Select in which WSL distro you want to mount the container
               </Typography>
-            }
-            {isPullingImage &&
-              <>
-                <Typography>
-                  Pulling localstack/localstack:latest
-                </Typography>
-                <DownloadProgress
-                  imageName={LATEST_IMAGE}
-                  callback={endOfDownloadCallback} />
-              </>
-            }
-            {
-              (hasLocalImage.checking || userState.loading) && <CircularProgress />
-            }
+              {
+                osState.loading ?
+                  <ShrinkedCircularProgress/> 
+                  :
+                  <FormControl sx={{ minWidth: 120 }} size="small" variant='outlined'>
```

Comment:
nit: looks like we're using a mix of single and double quotes in JSX elements - could be an opportunity to unify the styles? (again, a linter could take care of that..)